File: 021.py
#!/usr/bin/env python
#-*- coding:utf-8 -*-
from euler import proper_divisor_sum

ans = 0
for i in xrange(2, 10000):
    j = proper_divisor_sum(i)
    if i != j and proper_divisor_sum(j) == i:
        ans += j

print(ans)

File: 021.rb
#!/usr/bin/env ruby
# encoding: utf-8
require_relative "euler"

ans = 0
(2 .. 1e4).each do |x|
  y = x.proper_divisor_sum
  if x != y && y.proper_divisor_sum == x
    ans += y
  end
end

puts ans
